packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,2841 @@
+            return;
+          }
+
+          if (hovering) {
+            controller.open();
+            controller._anchor!._focusButton();
+          }
+        }
+
+        return TextButton(
+          style: mergedStyle,
+          focusNode: _buttonFocusNode,
+          onHover: _enabled ? (bool hovering) => handleHover(hovering, context) : null,
+          onPressed: _enabled ? () => toggleShowMenu(context) : null,
+          child: _MenuItemLabel(
+            leadingIcon: widget.leadingIcon,
+            trailingIcon: widget.trailingIcon,
+            hasSubmenu: true,
+            showDecoration: (controller._anchor!._parent?._orientation ?? Axis.horizontal) == Axis.vertical,
+            child: child ?? const SizedBox(),
+          ),
+        );
+      },
+      menuChildren: widget.menuChildren,
+      child: widget.child,
+    );
+  }
+
+  void _handleFocusChange() {
+    if (_buttonFocusNode.hasPrimaryFocus) {
+      if (!_menuController.isOpen) {
+        _menuController.open();
+      }
+    } else {
+      if (!_menuController._anchor!._menuScopeNode.hasFocus && _menuController.isOpen) {
+        _menuController.close();
+      }
+    }
+  }
+}
+
+/// The type of builder function used by [MenuAnchor.builder] to build the
+/// widget that the [MenuAnchor] surrounds.
+///
+/// The `context` is the context that the widget is being built in.
+///
+/// The `controller` is the [MenuController] that can be used to open and close
+/// the menu with.
+///
+/// The `child` is an optional child supplied as the [MenuAnchor.child]
+/// attribute. The child is intended to be incorporated in the result of the
+/// function.
+typedef MenuAnchorChildBuilder = Widget Function(
+  BuildContext context,
+  MenuController controller,
+  Widget? child,
+);
+
+/// A widget used to mark the "anchor" for a set of submenus, defining the
+/// region used to position the menu, which can be done either with an explicit
+/// location, or with an alignment.
+///
+/// When creating a menu with [MenuBar] or a [SubmenuButton], a [MenuAnchor] is
+/// not needed, since they provide their own internally.
+///
+/// The [MenuAnchor] is meant to be a slightly lower level interface than
+/// [MenuBar], used in situations where a [MenuBar] isn't appropriate, or to
+/// construct widgets or screen regions that have submenus.
+///
+/// {@tool dartpad}
+/// This example shows how to use a [MenuAnchor] to wrap a button and open a
+/// cascading menu from the button.
+///
+/// ** See code in examples/api/lib/material/menu_bar/menu_anchor.0.dart **
+/// {@end-tool}
+///
+/// {@tool dartpad}
+/// This example shows how to use a [MenuAnchor] to create a cascading context
+/// menu in a region of the view, positioned where the user clicks the mouse
+/// with Ctrl pressed.
+///
+/// ** See code in examples/api/lib/material/menu_bar/menu_anchor.1.dart **
+/// {@end-tool}
+class MenuAnchor extends StatefulWidget with DiagnosticableTreeMixin {
```

Comment:
I am struggling a little bit to justify the three different constructors that this class offers....

* `positioned` is sort of strange because the menu anchor doesn't really have an argument to position the menu. The position is passed to the controller - and that hopefully also works when the other constructors are used? Also, it magically disables the tap region...
* `bar` is completly different because it actually renders UI instead of just being an anchor...